### PR TITLE
feat(cli): add a 'navidrome plugin' CLI for managing and inspecting plugins

### PR DESCRIPTION
**Pull request by deluan**
### Description

This PR adds a `navidrome plugin` command-line interface for managing and inspecting plugins, filling a gap where plugin administration was only possible through the Native API, the Web UI, or hand-editing the plugins folder.

The command works **directly in-process** — it opens the database and plugin manager the same way the existing `user`, `backup`, and `scan` commands do (`getAdminContext` + `GetPluginManager`), so it needs no running server and no credentials. It is a thin shell over the **same `plugins.Manager` methods the Native API already calls**, so the CLI, API, and UI stay behaviorally identical.

Subcommands:

| Command | Purpose |
|---|---|
| `plugin list` | List installed plugins (`-f table\|csv\|json`) |
| `plugin info <id\|file.ndp>` | Show details for an installed plugin or an on-disk `.ndp` package (`-f text\|json`) |
| `plugin enable` / `disable <id>` | Enable/disable a plugin |
| `plugin edit <id>` | Atomically update config and/or permissions |
| `plugin rescan` | Re-discover plugins in the plugins folder |
| `plugin validate <id\|file.ndp>` | Validate an installed plugin or a `.ndp` package manifest |

Design notes:

- **Off-disk inspection** — `info` and `validate` accept either an installed plugin ID or a `.ndp` file path (auto-detected by the `.ndp` suffix + file existence; exploded folders are intentionally not supported). The off-disk path never touches the database and works even when the plugin system is disabled, making it a useful developer/CI pre-flight tool. All other commands require `Plugins.Enabled` and fail fast otherwise.
- **`edit` mirrors the Native API's read-modify-write semantics** — it loads the current plugin first and only overwrites the fields you explicitly set, so flipping one flag (e.g. `--write-access`) never wipes unspecified fields. It also validates that `--users`/`--libraries` values are JSON and that config passes the plugin's schema before persisting.
- A couple of small helpers in the `plugins` package were exported for reuse (`ReadManifest`, `ComputeFileSHA256`) and `Permissions.DeclaredNames()` was added as the single source of truth for permission names.

Scope is intentionally limited to **managing/inspecting plugins that already exist on disk** — there is no package installer/downloader.

### Type of Change

- [ ] Bug fix
- [x] New feature
- [ ] Documentation update
- [x] Refactor
- [ ] Other (please describe):

### Checklist

- [x] My code follows the project's coding style
- [x] I have tested the changes locally
- [ ] I have added or updated documentation as needed
- [x] I have added tests that prove my fix/feature works (or explain why not)
- [x] All existing and new tests pass

### How to Test

Plugin management is admin-scoped and requires the plugin system enabled. The off-disk `info`/`validate` paths work without a running server.

1. **Set up an environment with plugins enabled** and an admin user (e.g. `Plugins.Enabled=true`, a data folder, and at least one admin). Drop a `.ndp` package into the plugins folder (the repo's `plugins/testdata/test-config.ndp` works).
2. **List:**
   ```
   navidrome plugin list
   navidrome plugin list -f json
   ```
   Expect the installed plugin to appear; `-f json`/`-f csv` produce machine-readable output.
3. **Inspect (installed and off-disk):**
   ```
   navidrome plugin info test-config
   navidrome plugin info /path/to/test-config.ndp
   navidrome plugin info /path/to/test-config.ndp -f json
   ```
   Off-disk shows the manifest summary, declared permissions, and sha256 without touching the DB.
4. **Enable / disable round-trip:**
   ```
   navidrome plugin enable test-config && navidrome plugin list -f csv   # enabled,true
   navidrome plugin disable test-config && navidrome plugin list -f csv  # enabled,false
   ```
5. **Edit (config + permissions):**
   ```
   navidrome plugin edit test-config --config '{"api_key":"secret"}'
   navidrome plugin edit test-config --libraries '[1,2]' --write-access
   navidrome plugin edit test-config --no-write-access   # libraries [1,2] preserved
   ```
   Verify with `navidrome plugin info test-config`. Edge cases that should fail clearly: invalid config JSON / schema, `--users alice,bob` (non-JSON), no flags at all, and `--users x --all-users` (mutually exclusive).
6. **Validate:**
   ```
   navidrome plugin validate /path/to/test-config.ndp   # "<path>: OK"
   ```
   A malformed/invalid `.ndp` exits non-zero with a descriptive error.
7. **Guard / off-disk exception:** with `Plugins.Enabled=false`, `list`/`enable`/`rescan` fail with "Plugin system is disabled", while off-disk `validate`/`info` of a `.ndp` still work.

### Additional Notes

- **No new public API or UI surface, and no changes to existing routes** — the CLI reuses the existing `plugins.Manager` interface. The diff touches only `cmd/plugin.go`, a few exported helpers in `plugins/`, and tests.
- **Concurrency caveat:** like the existing `user`/`backup` CLI commands, mutating subcommands open the SQLite database directly. SQLite is single-writer, so running them while the server is live can briefly contend on the DB. This matches the behavior of the other CLI commands and is documented behavior, not new.
- This branch contains some follow-up refactor/test-cleanup commits (exporting helpers over thin wrappers, consolidating test files) made while polishing the feature; they're behavior-preserving.

### CODE REVIEW

File: cmd/plugin.go [hunk 1]
```diff
@@ -0,0 +1,507 @@
...
+
+	pluginEditCmd.Flags().StringVar(&editConfig, "config", "", "plugin config as JSON")
+	pluginEditCmd.Flags().StringVar(&editConfigFile, "config-file", "", "read plugin config JSON from a file ('-' for stdin)")
+	pluginEditCmd.MarkFlagsMutuallyExclusive("config", "config-file")
+	pluginEditCmd.Flags().StringVar(&editUsers, "users", "", "comma-separated usernames the plugin may access")
+	pluginEditCmd.Flags().BoolVar(&editAllUsers, "all-users", false, "grant the plugin access to all users")
+	pluginEditCmd.MarkFlagsMutuallyExclusive("users", "all-users")
+	pluginEditCmd.Flags().StringVar(&editLibraries, "libraries", "", "comma-separated library IDs the plugin may access")
+	pluginEditCmd.Flags().BoolVar(&editAllLibs, "all-libraries", false, "grant the plugin access to all libraries")
+	pluginEditCmd.MarkFlagsMutuallyExclusive("libraries", "all-libraries")
```
[the remark this answers is not in the document]
Comment (deluan) [reply cmd/plugin.go]: Fixed in 3d2db6a9. The stored format is JSON (the UI sends JSON.stringify and the manager json.Unmarshals these fields), so I kept JSON support but also added comma-separated input (converted to the JSON array the manager expects). Help text now documents both.

File: cmd/plugin.go [hunk 2]
```diff
@@ -0,0 +1,507 @@
...
+		if opts.allLibraries != nil {
+			allLibs = *opts.allLibraries
+		}
+		if opts.writeAccess != nil {
+			writeAccess = *opts.writeAccess
+		}
+		if err := mgr.UpdatePluginLibraries(ctx, id, libs, allLibs, writeAccess); err != nil {
+			return err
+		}
+	}
```
[the remark this answers is not in the document]
Comment (deluan) [reply cmd/plugin.go:491]: Adopted in 3d2db6a9 — --users/--libraries now accept comma-separated input or a JSON array (value starting with '[' is treated as JSON). Implemented via usersToJSON/librariesToJSON helpers; non-integer library IDs are rejected with a clear error.

File: cmd/plugin.go [hunk 3]
```diff
@@ -0,0 +1,507 @@
...
+	pluginEditCmd.MarkFlagsMutuallyExclusive("config", "config-file")
+	pluginEditCmd.Flags().StringVar(&editUsers, "users", "", "comma-separated usernames the plugin may access")
+	pluginEditCmd.Flags().BoolVar(&editAllUsers, "all-users", false, "grant the plugin access to all users")
+	pluginEditCmd.MarkFlagsMutuallyExclusive("users", "all-users")
+	pluginEditCmd.Flags().StringVar(&editLibraries, "libraries", "", "comma-separated library IDs the plugin may access")
+	pluginEditCmd.Flags().BoolVar(&editAllLibs, "all-libraries", false, "grant the plugin access to all libraries")
+	pluginEditCmd.MarkFlagsMutuallyExclusive("libraries", "all-libraries")
+	pluginEditCmd.Flags().BoolVar(&editWriteAccess, "write-access", false, "allow the plugin write access to libraries")
+	pluginEditCmd.Flags().BoolVar(&editNoWrite, "no-write-access", false, "deny the plugin write access to libraries")
+	pluginEditCmd.MarkFlagsMutuallyExclusive("write-access", "no-write-access")
```
[the remark this answers is not in the document]
Comment (deluan) [reply cmd/plugin.go:68]: cobra's MarkFlagsMutuallyExclusive has signature 'func (c *Command) MarkFlagsMutuallyExclusive(flagNames ...string)' — it returns nothing, so there is no error to handle. This matches existing usage in cmd/user.go. No change.

File: plugins/package_test.go [hunk 4]
```diff
@@ -132,66 +132,55 @@ var _ = Describe("ndpPackage", func() {
...
 			manifest := &Manifest{
 				Name:        "Test Plugin",
 				Author:      "Test Author",
 				Version:     "1.0.0",
 				Description: new("A test plugin"),
 			}
-			wasmBytes := make([]byte, 1024*1024) // 1MB of zeros
 
-			err := createTestPackage(ndpPath, manifest, wasmBytes)
+			err := createTestPackage(ndpPath, manifest, nil)
```
[the remark this answers is not in the document]
Comment (deluan) [reply plugins/package_test.go:145]: ReadManifest returns only (*Manifest, error); it has no way to expose whether the wasm bytes were read, so a payload size can't assert 'manifest-only' behavior. That's why the spec was renamed to what it actually verifies (manifest parses from a package containing a wasm entry). Restoring 1MB would only add an allocation that nothing asserts on. No change.

File: plugins/package_test.go [hunk 5]
```diff
@@ -132,66 +132,55 @@ var _ = Describe("ndpPackage", func() {
...
-		It("should return error for missing manifest", func() {
-			ndpPath := filepath.Join(tmpDir, "no-manifest.ndp")
-
-			f, err := os.Create(ndpPath)
-			Expect(err).ToNot(HaveOccurred())
-			defer f.Close()
+		It("returns an error for a non-existent file", func() {
+			_, err := ReadManifest(filepath.Join(tmpDir, "does-not-exist.ndp"))
+			Expect(err).To(HaveOccurred())
+		})
```
[the remark this answers is not in the document]
Comment (deluan) [reply plugins/package_test.go:157]: Good catch — openPackage and ReadManifest have separate loops, each with its own 'package missing manifest.json' return, so ReadManifest's branch was uncovered. Restored a ReadManifest missing-manifest test asserting that error in 3d2db6a9.

File: plugins/manifest.go [hunk 6]
```diff
@@ -3,10 +3,39 @@ package plugins
...
+		"cache":       p.Cache != nil,
+		"http":        p.Http != nil,
+		"kvstore":     p.Kvstore != nil,
+		"library":     p.Library != nil,
+		"scheduler":   p.Scheduler != nil,
+		"subsonicapi": p.Subsonicapi != nil,
+		"taskqueue":   p.Taskqueue != nil,
+		"users":       p.Users != nil,
+		"websocket":   p.Websocket != nil,
+	} {
```
[the remark this answers is not in the document]
Comment (deluan) [reply plugins/manifest.go]: Fixed in 3d2db6a9 — DeclaredNames now reflects over the Permissions struct's json tags (which are generated from manifest-schema.json), so new permission types are picked up automatically instead of via a hand-maintained list.

File: cmd/plugin.go [hunk 7]
```diff
@@ -0,0 +1,507 @@
...
+	}
+)
+
+func isPackagePath(arg string) bool {
+	if !strings.HasSuffix(arg, plugins.PackageExtension) {
+		return false
+	}
+	info, err := os.Stat(arg)
+	return err == nil && !info.IsDir()
+}
```
[the remark this answers is not in the document]
Comment (deluan) [reply cmd/plugin.go:145]: Done in 3d2db6a9 — isPackagePath now checks only the .ndp suffix, so a mistyped path routes to ReadManifest and reports a precise 'no such file' error.

File: cmd/plugin_test.go [hunk 8]
```diff
@@ -0,0 +1,313 @@
...
+		Expect(mgr.UpdatePluginUsersCalls[0].UsersJSON).To(Equal(`["bob"]`)) // not wiped
+	})
+
+	It("rejects a non-JSON users value", func() {
+		mgr := &tests.MockPluginManager{}
+		users := "alice,bob"
+		err := applyPluginEdit(context.Background(), mgr, cur, pluginEditOptions{users: &users})
+		Expect(err).To(HaveOccurred())
+		Expect(mgr.UpdatePluginUsersCalls).To(BeEmpty())
+	})
```
[the remark this answers is not in the document]
Comment (deluan) [reply cmd/plugin_test.go]: Added in 3d2db6a9 — specs cover comma-separated users, comma-separated library IDs (parsed to ints), JSON passthrough, malformed JSON, and a non-integer library ID error.

File: cmd/plugin.go [hunk 9]
```diff
@@ -0,0 +1,507 @@
...
+	if opts.config != nil {
+		if err := mgr.ValidatePluginConfig(ctx, id, *opts.config); err != nil {
+			return fmt.Errorf("invalid config: %w", err)
+		}
+		if err := mgr.UpdatePluginConfig(ctx, id, *opts.config); err != nil {
+			return err
+		}
+	}
+	if opts.users != nil || opts.allUsers != nil {
+		users, allUsers := cur.Users, cur.AllUsers
```
[the remark this answers is not in the document]
Comment (deluan) [reply cmd/plugin.go:456]: Fixed in 83624deef — setting an explicit --users list now implies allUsers=false (an allow-list means 'restrict to these users'), so it's no longer silently ignored when the plugin was set to all-users. Verified end-to-end.

File: cmd/plugin.go [hunk 10]
```diff
@@ -0,0 +1,507 @@
...
+		}
+		if opts.allUsers != nil {
+			allUsers = *opts.allUsers
+		}
+		if err := mgr.UpdatePluginUsers(ctx, id, users, allUsers); err != nil {
+			return err
+		}
+	}
+	if opts.libraries != nil || opts.allLibraries != nil || opts.writeAccess != nil {
+		libs, allLibs, writeAccess := cur.Libraries, cur.AllLibraries, cur.AllowWriteAccess
```
[the remark this answers is not in the document]
Comment (deluan) [reply cmd/plugin.go:473]: Fixed in 83624deef — same treatment for libraries: an explicit --libraries list now implies allLibraries=false.
